hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -383,4 +369,55 @@ private void handlePutOMKeyEvent(String key, OmKeyInfo omKeyInfo,
     }
   }
 
+  /**
+   * Write an OM key to container DB and update containerID -> no. of keys
+   * count.
+   *
+   * @param key key String
+   * @param omKeyInfo omKeyInfo value
+   * @throws IOException if unable to write to recon DB.
+   */
+  private void writeOMKeyToContainerDB(String key, OmKeyInfo omKeyInfo)
+      throws IOException {
+    long containerCountToIncrement = 0;
+    for (OmKeyLocationInfoGroup omKeyLocationInfoGroup : omKeyInfo
+        .getKeyLocationVersions()) {
+      long keyVersion = omKeyLocationInfoGroup.getVersion();
+      for (OmKeyLocationInfo omKeyLocationInfo : omKeyLocationInfoGroup
+          .getLocationList()) {
+        long containerId = omKeyLocationInfo.getContainerID();
+        ContainerKeyPrefix containerKeyPrefix = new ContainerKeyPrefix(
+            containerId, key, keyVersion);
+        if (reconContainerMetadataManager.getCountForContainerKeyPrefix(
+            containerKeyPrefix) == 0) {
+          // Save on writes. No need to save same container-key prefix
+          // mapping again.
+          reconContainerMetadataManager.storeContainerKeyMapping(
+              containerKeyPrefix, 1);
+
+          // check if container already exists and
+          // increment the count of containers if it does not exist
+          if (!reconContainerMetadataManager.doesContainerExists(containerId)) {
+            containerCountToIncrement++;
+          }
+
+          // update the count of keys for the given containerID
+          long keyCount = reconContainerMetadataManager
+              .getKeyCountForContainer(containerId);
```

Comment:
getKeyCountForContainer also we can avoid querying from rocksdb always, Initialize keyCount as 0 and if container doesn't exist we can avoid calling this.

Reply:
Following the previous conversation, we can follow the implementation in `handlePutOMKeyEvent` function using the `containerKeyCountMap` instead, as follows:

```
// update the count of keys for the given containerID
long keyCount;
if (containerKeyCountMap.containsKey(containerId)) {
  keyCount = containerKeyCountMap.get(containerId);
} else {
  keyCount = reconContainerMetadataManager
      .getKeyCountForContainer(containerId);
}

// increment the count and update containerKeyCount.
// keyCount will be 0 if containerID is not found. So, there is no
// need to initialize keyCount for the first time.
containerKeyCountMap.put(containerId, ++keyCount);
```

If the container has been encountered before, we can use the look up from `containerKeyCountMap`.

Reply:
@ashishkumar50 I have addressed the suggestion, PTAL, thank you.